flexget/plugins/list/sonarr_list.py

```diff
@@ -37,69 +41,23 @@ class SonarrSet(MutableSet):
             'series_type': {'type': 'string', 'enum': ['standard', 'daily', 'anime'], 'default': 'standard'},
             'tags': {'type': 'array', 'items': {'type': 'integer'}, 'default': [0]}
         },
-        'required': ['api_key', 'base_url'],
+        'required': ['api_key'],
         'additionalProperties': False
     }
 
-    def series_request_builder(self, base_url, port, api_key):
-        parsedurl = urlparse(base_url)
-        log.debug('Received series list request')
-        url = '%s://%s:%s%s/api/series' % (parsedurl.scheme, parsedurl.netloc, port, parsedurl.path)
-        headers = {'X-Api-Key': api_key}
-        return url, headers
-
-    def lookup_request(self, base_url, port, api_key):
-        parsedurl = urlparse(base_url)
-        log.debug('Received series lookup request')
-        url = '%s://%s:%s%s/api/series/lookup?term=' % (parsedurl.scheme, parsedurl.netloc, port, parsedurl.path)
-        headers = {'X-Api-Key': api_key}
-        return url, headers
-
-    def profile_list_request(self, base_url, port, api_key):
-        parsedurl = urlparse(base_url)
-        log.debug('Received profile list request')
-        url = '%s://%s:%s%s/api/profile' % (parsedurl.scheme, parsedurl.netloc, port, parsedurl.path)
-        headers = {'X-Api-Key': api_key}
-        return url, headers
-
-    def rootfolder_request(self, base_url, port, api_key):
-        parsedurl = urlparse(base_url)
-        log.debug('Received rootfolder list request')
-        url = '%s://%s:%s%s/api/Rootfolder' % (parsedurl.scheme, parsedurl.netloc, port, parsedurl.path)
-        headers = {'X-Api-Key': api_key}
-        return url, headers
-
-    def get_json(self, url, headers):
+    def _sonarr_request(self, endpoint, term=None, method='get', data=None):
+        base_url = self.config['base_url']
+        port = self.config['port']
+        url = '{}:{}/api/{}'.format(base_url, port, endpoint)
+        headers = {'X-Api-Key': self.config['api_key']}
+        if term:
+            url += '?term={}'.format(term)
         try:
-            response = requests.get(url, headers=headers)
-            if response.status_code == 200:
-                return response.json()
-            else:
-                raise plugin.PluginError('Invalid response received from Sonarr: %s' % response.content)
+            rsp = requests.request(method, url, headers=headers, json=data)
+            rsp.raise_for_status()
```

Comment:
Why not use https://github.com/Flexget/Flexget/blob/develop/flexget/utils/requests.py#L214 ?

Reply:
Dunno. Since I didn't need to use our custom session features I saw no reason using our own implementation

Reply:
It's just a wrapper. I'd rather we were consistent on the matter. You can just use this https://github.com/Flexget/Flexget/blob/develop/flexget/utils/requests.py#L253